### ISSUE TITLE
Chef 13: zap deletes everything

### ISSUE THREAD

**Issue by autarchprinceps** [nvwls/zap#42]
I recently tested updating my cookbook to run on Chef 13. When doing so a zap_directory deletes everything, rather than the cleanup behaviour it had with the otherwise same code in Chef 12.

The exact entry would be the following: 
``` ruby
zap_directory '/etc/sensu/conf.d/checks' do
  pattern '*.json'
end
```
This is supposed to clean up checks not created by chef, but it instead wipes everything in that folder. As mentioned it only does so in Chef 13, if I restrict it to use Chef 12 it operates correctly.

I'm using zap in version 0.15.1.

**Comment by lamont-granquist** [comment]
this is likely a dup of #35.

my guess is that starting with chef-13 use_inline_resources is the default behavior of chef resources, so the sensu cookbook was most likely using a resource that failed to declare that and now it is (correctly) getting a sub-resource collection.

**Comment by autarchprinceps** [comment]
I'm not sure I understand what you mean exactly. I'm creating sensu checks, which are json files in the /etc/sensu/conf.d/checks folder. This is not part of a sensu cookbook. The problem is that if a file is no longer set to be created in chef, that is not equivalent to being deleted. If my logic changes and I no longer create a check, that means I don't want that check, but in the way that chef handles it, it would continue to exist in the last form it was created.
This is where zap comes in handy for me. Under chef 12 it deletes all files not currently handled by chef, which are exactly those checks I no longer wish to have.

**Comment by lamont-granquist** [comment]
how are you creating those files?  because if you're using file resources in the outer recipe run_context and zap broke on chef 13 that would be unusual.

if you are using a helper resource that either you wrote or from the sensu cookbook which wraps a file resource that drops the json file into `/etc/sensu/conf.d/checks` then its likely exactly what i mentioned.  but you haven't provided enough information about your run to determine if that is the root cause of the problem or not.  my guess is that it is, because i don't know of anything that would have changed in chef-13 so that zap couldn't find a file resource in the outer run context -- that would be highly surprising.  but the change to make use_inline_resources the default would cause more issues due to the problem in #35 (which will keep on getting worse).

**Comment by nvwls** [comment]
Fixed in v1.0.0
